hw/ip/entropy_src/data/entropy_src.hjson

```diff
@@ -0,0 +1,153 @@
+        { bits: "31:0",
+          name: "ENTROPY_SRC Min-entropy",
+          desc: '''Read of this register provides the expected value of assessed min-etropy, 
+                 expressed as a single precision floating point number. This value is 1-x, 
+                 where x is the entropy per bit. For example, a full entropy source, as defined 
+                 by NIST 800-90-B, would have an x value of less than 2**(-64). This number 
+                 should be given as a lower bound for the entropy generated under normal operation.
+                 ''',
+          resval: "0x123ab"
+        }
+      ]
+    },
+    { name: "ESNOISE",
+      desc: "ENTROPY_SRC noise source",
+      swaccess: "ro",
+      hwaccess: "hrw",
+      hwext: "true",
+      hwqe: "true",
+      fields: [
+        { bits: "31:0" 
+          desc: "A read of this register provides noise needed for an entropy source."
+        }
+      ]
+    },
+    { name: "ESCTRL",
+      desc: "ENTROPY_SRC control register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          name: "Initialize ENTROPY_SRC",
+          desc: "Setting this bit will initialize the ENTROPY_SRC function."
+        }
+        { bits: "1",
+          name: "Lock Seed",
+          desc: "Setting this bit will not allow the LFSR seed input to be changed until a chip reset has occurred."
+        }
+      ]
+    },
+    { name:     "ESSTATUS"
+      desc:     "ENTROPY_SRC status register"
+      swaccess: "ro"
+      hwaccess: "hrw"
+      fields: [
+        { bits: "0"
+          name: "Entropy ready"
+          desc: '''This is set when the noise source FIFO has met or exceeded the ESTHRESHOLD value. This status bit will
+                automatically clear once the FIFO depth drops below the threshold value. This bit will not set unless the
+                block is enabled.
+                '''
+        }
+      ]
+    },
+    { name:     "ESFDEPTHST"
+      desc:     "ENTROPY_SRC FIFO depth status register"
+      swaccess: "ro"
+      hwaccess: "hrw"
+      fields: [
+        { bits: "6:0"
+          name: "Noise FIFO depth"
+          desc: "This register will reflect the current depth of the noise FIFO register."
+        }
+      ]
+    },
+    { name: "ESTHRESH",
+      desc: "ENTROPY_SRC threshold register",
+      swaccess: "rw",
+      hwqe: "true",
+      fields: [
+        { bits: "6:0",
+          desc: "The noise FIFO depth will be compared against this register value when setting the status bit and valid interrupt.",
+       }
+      ]
+    }
+    { name: "ESRATE",
+      desc: "ENTROPY_SRC bit rate register",
+      swaccess: "rw",
+      hwqe: "true",
+      fields: [
+        { bits: "15:0",
+          desc: '''This register determines the rate that noise bits will be supplied. It's purpose to is simulate slow noise sources.
+                A value of 1 will cut a full noise source in half. A value of 8 will cut the rate by 1/8th.
```

Comment:
The way I read it, it is doing the right thing. If you want one new word every 1000 cycles you set `ESRATE` to 1000. Perhaps it just needs to be described cleaner?

Reply:
This has been corrected and clarified. Thanks @mwbranstad .

Personally, I would find it better if a value of 1 would lead to full rate, and a value of 8 to 1/8 of the rate, instead of 1/2 and 1/9, respectively. But this is just my opinion.

Reply:
I see your point on this one. If changed, then need to determine what to do with the value of zero now - off? Since this is more of a modeling register the implementation is very simple, but will think about this some more.

Reply:
Decided to change this tothe suggested way, with 1 representing full rate. This give a new feature of zero turning noise generation off.

Reply:
The register could for example be made to reset to 1 and to turn a write of 0 (illegal) to writing 1. We also do this for AES e.g. when writing an illegal value to the key length config register.